protocol/contracts/templegold/TempleGoldVesting.sol

```diff
@@ -0,0 +1,339 @@
+pragma solidity ^0.8.20;
+// SPDX-License-Identifier: AGPL-3.0-or-later
+// Temple (templegold/TempleGoldVesting.sol)
+
+
+import { TempleElevatedAccess } from "contracts/v2/access/TempleElevatedAccess.sol";
+import { CommonEventsAndErrors } from "contracts/common/CommonEventsAndErrors.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { EnumerableSet } from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { TempleMath } from "contracts/common/TempleMath.sol";
+import { ITempleGoldVesting } from "contracts/interfaces/templegold/ITempleGoldVesting.sol";
+import { PaymentBase } from "contracts/admin/PaymentBase.sol";
+
+
+/**
+ * @title TGLD Vesting
+ * @notice Vesting contract for contributors. Token is TGLD and allocations are set with createSchedules.
+ *  An account can have multiple vesting schedules. Vesting schedules can be revoked and canceled.
+ */
+contract TempleGoldVesting is ITempleGoldVesting, PaymentBase, TempleElevatedAccess {
+    using SafeERC20 for IERC20;
+    using EnumerableSet for EnumerableSet.Bytes32Set;
+
+    /// @notice Total TGLD vested and unclaimed
+    uint256 public override totalVestedAndUnclaimed;
+    /// @notice Schdules for recipients
+    mapping(bytes32 id => VestingSchedule schedule) public schedules;
+      /// @notice Recipient vesting counts for generating IDs. An account can have multiple vesting schedules
+    mapping(address => uint256) public holdersVestingCount;
+    /// @notice Vesting Ids
+    EnumerableSet.Bytes32Set private _vestingIds;
```

Comment:
I wonder if this should be renamed to `_activeVestingIds`

Since when they get revoked they get removed from here (and he order of the items in the enumerable set will change from that)